### PR TITLE
Remove tmhoauth & replace with Guzzle

### PR DESCRIPTION
**Pull request by atymic**
First big PR for 3.x - remove tmhoauth.

Very WIP, if you are keen to help let me know :)

### CODE REVIEW

**Comment by reliq** [comment]
@atymic I'm willing to look into this.

It seems this also has namespace changes here.

Could you please outline the scope of this here.

**Comment by atymic** [comment]
The outline is on the main readme here:
https://github.com/atymic/twitter#3x

**Comment by atymic** [comment]
Maybe we can merge this to the 3.x branch and then we can PR against that.

File: src/Twitter.php [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+
+    private const DEFAULT_EXTENSION = 'json';
+    private const REQUEST_METHOD_GET = 'GET';
+    private const REQUEST_METHOD_POST = 'POST';
+    private const URL_FORMAT = 'https://%s/%s/%s.%s';
+    private const KEY_FORM_PARAMS = 'form_params';
+    private const KEY_QUERY = 'query';
+    private const KEY_FORMAT = 'format';
+
+    private const RESPONSE_CODE_400 = 400;
```
Comment (atymic) [inline src/Twitter.php]: I don't think it's worth adding these constant. It's just overhead over defining them in the switch inline and they aren't going to change.

File: src/Twitter.php [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+    /**
+     * @param Configuration $config
+     *
+     * @throws InvalidArgumentException
+     *
+     * @return HttpClient
+     */
+    private function getHttpClient(Configuration $config): HttpClient
+    {
+        $stack = HandlerStack::create();
```
Comment (atymic) [inline src/Twitter.php]: Great job with this :)

File: src/Twitter.php [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+        $body = (string)$response->getBody();
+
+        switch ($format) {
+            case self::RESPONSE_FORMAT_JSON:
+                return $body;
+            case self::RESPONSE_FORMAT_ARRAY:
+                return json_decode($body, true);
+            case self::RESPONSE_FORMAT_OBJECT:
+            default:
+                return json_decode($body, false);
```
Comment (atymic) [inline src/Twitter.php]: Do we want to keep the default as object? TBH would prefer it as array unless we are going to create DTOs for the responses.

Comment (reliq) [reply src/Twitter.php]: Either is fine actually. My only argument for `object` is that it was this by default before.

Comment (atymic) [reply src/Twitter.php]: Fair point. Happy to leave it as in for the moment.

File: src/Twitter.php (hunk 1 above)
Comment (reliq) [reply src/Twitter.php]: Usually, I prefer constants to magic numbers. In this case I guess it's fine.

**Pull request merged**